tests/models/test_ModelUtils.py

```diff
@@ -573,6 +574,35 @@ def test_get_named_leaf_modules() -> None:
     assert utils.get_named_leaf_modules(sequential2) == {"0.0": linear1, "0.1": linear2}
 
 
+@pytest.mark.parametrize(
+    "strategy, expected_fn",
+    [
+        ("learn", "initialize_learnable_positional_embedding"),
+        ("sincos", "initialize_2d_sine_cosine_positional_embedding"),
+        ("skip", None),
+    ],
+)
+def test_initialize_positional_embedding(
+    strategy: str, expected_fn: Optional[str], mocker: MockerFixture
+) -> None:
+    if expected_fn is not None:
+        mock_fn = mocker.spy(utils, expected_fn)
+    pos_embedding = Parameter(torch.rand(1, 1, 64))
+    utils.initialize_positional_embedding(
+        pos_embedding=pos_embedding, strategy=strategy, num_prefix_tokens=1
+    )
+    if expected_fn is not None:
+        mock_fn.assert_called_once()
+
+
+def test_initialize_learnable_positional_embedding() -> None:
+    pos_embedding = Parameter(torch.ones(1, 1, 64) * 10)
+    utils.initialize_learnable_positional_embedding(pos_embedding)
+    assert pos_embedding.requires_grad
+    # Trunc normal should clip values to small size.
+    assert torch.all(pos_embedding.abs() < 5)
```

Comment:
Why not assert that [init.trunc_normal](https://pytorch.org/docs/stable/nn.init.html#torch.nn.init.trunc_normal_) was called?

If you want to keep it like this, I would rather assert that all are <=2, as this is the default value.

Reply:
I want to avoid mocking if not necessary. I changed now the test to just check that the embedding was updated, this avoids relying on implementation details of how the embedding is initialized.